### PR TITLE
[Intl][String] Document emoji

### LINKED ISSUE #17191: [Intl] Update EmojiTransliterator to translate emoji to github and slac…
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#47263
| PR author(s) | @lyrixx
| Merged in    | 6.2

### LINKED ISSUE #17193: [String] Add support for emoji in AsciiSlugger
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#47264
| PR author(s) | @lyrixx
| Merged in    | 6.2

### PR DESCRIPTION
**Pull request by lyrixx**
fixes #17191
fixes #17193

### CODE REVIEW

File: components/intl.rst [hunk 1]
```diff
@@ -385,9 +385,25 @@ their textual representation in all languages based on the `Unicode CLDR dataset
...
+The ``EmojiTransliterator`` class also provides two extra catalogues: ``github``
+and ``slack`` that converts any emojis to the corresponding short code in the
+respective platforms::
+
+    use Symfony\Component\Intl\Transliterator\EmojiTransliterator;
+
+    // describe emojis in Slack short code
+    $transliterator = EmojiTransliterator::create('slack');
+    $transliterator->transliterate('Menus with 🍕 or 🍝');
+    // => 'Menus with :green_salad: or :falafel:'
```
Comment (94noni) [inline components/intl.rst:397]: As a lebanese, 🍝 are not 🧆 😁
Is this an emoji typo?

Comment (wouterj) [reply components/intl.rst:397]: Nor is a :pizza: :green_salad: :)

Comment (94noni) [reply components/intl.rst:397]: Hahaha

File: components/string.rst [hunk 2]
```diff
@@ -526,6 +519,36 @@ the injected slugger is the same as the request locale::
         }
     }
 
+.. _string-slugger-emoji:
+
+Slug emojis
```
Comment (OskarStark) [inline components/string.rst]:
```suggestion
Slug Emojis
```

File: components/string.rst [hunk 3]
```diff
@@ -526,6 +519,36 @@ the injected slugger is the same as the request locale::
...
+    $slugger = $slugger->withEmoji();
+
+    $slug = $slugger->slug('a 😺, 🐈‍⬛, and a 🦁 go to 🏞️', '-', 'en');
+    // $slug = 'a-grinning-cat-black-cat-and-a-lion-go-to-national-park';
+
+    $slug = $slugger->slug('un 😺, 🐈‍⬛, et un 🦁 vont au 🏞️', '-', 'fr');
+    // $slug = 'un-chat-qui-sourit-chat-noir-et-un-tete-de-lion-vont-au-parc-national';
+
+If you want to use a specific locale for the emoji, or to use the shorts code
+from `github` or `slack`, use the first argument of ``withEmoji()`` method:
```
Comment (OskarStark) [inline components/string.rst]:
```suggestion
from `github` or `slack`, use the first argument of ``withEmoji()`` method::
```

**Comment by OskarStark** [comment]
Please add versionadded directives, thank you

File: components/intl.rst (hunk 1 above)
Comment (lyrixx) [reply components/intl.rst:397]: Oups :) Fixed

**Review by lyrixx — COMMENTED** [review]
Thanks for the review, I addressed your comments

File: components/string.rst [hunk 4]
```diff
@@ -526,6 +519,39 @@ the injected slugger is the same as the request locale::
...
+.. _string-slugger-emoji:
+
+Slug Emojis
+~~~~~~~~~~~
+
+.. versionadded:: 6.2
+
+    The Emoji transliteration feature was introduced in Symfony 6.2.
+
+You can can transform any emojis into a textual representation::
```
Comment (xabbuh) [inline components/string.rst]:
```suggestion
You can transform any emojis into a textual representation::
```

Comment (lyrixx) [reply components/string.rst]: Thanks, fixed

**Review by OskarStark — APPROVED** [review]

**Review by xabbuh — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Nice feature! Thanks Grégoire!

**Pull request merged**